static/specification.html

```diff
@@ -138,605 +336,555 @@ <h2>
+        Task sources
+      </h2>
       <p>
-        A provider can be implemented as a Payment Handler supporting the
-        'webmonetization' payment method (The payment method specification for
-        this payment method is still under development as it is dependant on
-        changes to the Payment Handler API that support streaming payments).
-        Communication between the browser and the provider would use this flow.
+        The following [=task source=] is defined by this specifications.
       </p>
-      <ul>
-        <li>After parsing the `&lt;meta&gt;` Tags Set, the browser creates a
-        new <a href=
-        "https://www.w3.org/TR/payment-request/#paymentrequest-interface">PaymentRequest</a>
-        object with the following <a href=
-        "https://www.w3.org/TR/payment-request/#dom-paymentmethoddata">PaymentMethodData</a>
-        argument.
-        </li>
-      </ul>
-      <pre class="json">
-        {
-          "supportedMethods": "webmonetization",
-          "data": {
-            "paymentPointer": "&lt;payment pointer parsed from meta tag&gt;"
-          }
-        }
-      </pre>
-      <ul>
-        <li>The <a>user agent</a> calls <code>.show()</code> on this
-        PaymentRequest, triggering the <a href=
-        "https://www.w3.org/TR/payment-handler/">PaymentHandler</a> for <code>
-          webmonetization</code>. This PaymentHandler is how the browser
-          communicates to the provider.
-          <ul>
-            <li>The return value of <code>.show()</code> MUST return a Promise,
-            and must also implement the {{ EventTarget }} interface. The
-            provider will emit {{ MonetizationStartEvent }} and {{
-            MonetizationProgressEvent }} events from this Promise to
-            communicate to the browser when payment occurs. The Promise MUST
-            NOT resolve, because Web Monetization continues for the entire
-            lifetime of a given meta tag. The Promise MAY reject if there is an
-            error preventing the provider from paying and no retries will
-            occur.
-            </li>
-            <li>This PaymentHandler MUST NOT require any UI or confirmation to
-            proceed with payment.
-            </li>
-          </ul>
-        </li>
-      </ul>
+      <dl>
+        <dt>
+          The <dfn>monetization task source</dfn>
+        </dt>
+        <dd>
+          Used by this specification to queue up non-blocking
+          {{MonetizationEvent}}s.
+        </dd>
+      </dl>
     </section>
-    <section id="specification">
+    <section data-dfn-for="MonetizationEvent">
       <h2>
-        Specification
+        `MonetizationEvent` interface
       </h2>
-      <section id="meta-tags-set">
+      <pre class="idl">
+        [SecureContext, Exposed=Window]
+        interface MonetizationEvent : Event {
+          constructor(DOMString type, MonetizationEventInit eventInitDict);
+          readonly attribute DOMString amount;
```

Comment:
The amount of confusion/bugs we've had over the years forgetting whether the amount was a source or received amount. If we are to have an amount (not sure how it relates to an outgoing payment ), perhaps we should explicitly name it sourceAmount/destinationAmount etc. 

Why not both? 

I believe @adrianhopebailie  was arguing for sourceAmount only, so that the monetization event could be reported ASAP. If you reported both, you’d have to wait for the packet to be sent to get the (received)Amount.

Why not sourceAmount only?

As a website, you own the payment pointer, you know the destination currency, and you want to code around that.

Reply:
I argue that the purpose of the event is to **notify the website that a payment has been sent**, not that a payment has been received.

Assuming that we are able to notify the website that a payment has been received seems leaky and error prone (as we've discovered). It makes it very difficult to be flexible in the implementation of the payments behind this API.

Ultimately the website is going to do 1 of 2 things with this information:

1. Trust that the amount was sent since its so small and the good or service provided in return so low in value that it doesn't bother verifying the payment was received. In this case it either does a rudimentary conversion to an asset it knows, using some public exchange rate, or simply doesn't bother because it's not checking it received the funds anyway so what value does it get from converting the send amount to the receive currency.

2. It verifies the payment at the receive account and in so doing gets the receive amount in the currency of the receive account.

I suggest we stick with sendAmount in the event and leave it to the website to calculate or fetch the receive amount